src/main/java/racingcar/RacingCarGameLogic.java

```diff
@@ -17,15 +21,32 @@ public static RacingCarGameLogic getInstance() {
         return instance;
     }
 
-    public void gameLogic(int numberOfCar, int numberOfAttempt) {
-        List<Car> cars = initCars(numberOfCar);
-        IntStream.range(0, numberOfAttempt).forEach(i -> RacingCarGameRule.getInstance().moveCars(cars));
+    public void gameLogic(String nameOfCars, int numberOfAttempt) {
+        List<Car> cars = initCars(nameOfCars);
+
+        for (int i = 0; i < numberOfAttempt; i++) {
+            RacingCarGameRule.getInstance().moveCars(cars);
+            ResultView.getInstance().printMove(cars);
+            ResultView.getInstance().printLineBreak();
+        }
+
+        List<Car> winners = RacingCarGameRule.getInstance().getWinnerCars(cars);
+        ResultView.getInstance().printRacingcarWinners(winners);
     }
 
-    private List<Car> initCars(int numberOfCar) {
-        List<Car> cars = new ArrayList<>(numberOfCar);
-        IntStream.range(0, numberOfCar).forEach(i -> cars.add(new Car()));
+    public List<Car> initCars(String nameOfCars) {
+        String[] names = getCarNamesSplit(nameOfCars);
+        List<Car> cars = new ArrayList<>(names.length);
+
+        for (int i = 0; i < names.length; i++) {
+            cars.add(new Car(names[i],INIT_CAR_DISTANCE));
+        }
```

Comment:
```suggestion
        for (String name: names) {
            cars.add(new Car(name, INIT_CAR_DISTANCE));
        }
```
enhanced for loop 문법을 사용하면
인덱스를 명시할 필요가 없어집니다.

인덱스가 필요한 게 아니라면 거의 사용하지 않는 것 같아요